target/mips/op_helper_cheri.c

```diff
@@ -208,43 +208,45 @@ static target_ulong ccall_common(CPUArchState *env, uint32_t cs, uint32_t cb, ui
 {
     const cap_register_t *csp = get_readonly_capreg(env, cs);
     const cap_register_t *cbp = get_readonly_capreg(env, cb);
+
+    int allow_unsealed = (selector == CCALL_SELECTOR_2) && !cap_is_sealed_with_type(csp);
```

Comment:
It seems to me like you could end up with a sealed value in PCC?

Reply:
Ah, because I have used !cap_is_sealed_with_type (to mean unsealed) and csp might be sentry? It probably wants to read ((selector == 2) && cap_is_unsealed). I don't think this can currently go wrong because there would be no valid cbp to use in conjunction with a sentry csp.

Reply:
Yes, it needs to be unsealed if it's a sentry. I guess it's fine if idc is sealed if the code cap is a sentry.

Reply:
Used the correct call.